Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -68,31 +116,68 @@ namespace MultiplayerSample
 
     void NetworkWeaponsComponent::OnWeaponActivate([[maybe_unused]] const WeaponActivationInfo& activationInfo)
     {
+        // If we're replaying inputs then early out
```

Comment:
Is this because it's debug draw, because we expect replay to manage projectile life cycle or both?

Reply:
These callbacks will invoke during corrections if we replay inputs where weapon activations occurred. This means you shoot a weapon, get cool muzzle flash effects and such, immediately get a correction and your weapon would retrigger all those muzzle flash effects. Very bad visual bug. This guard here just ensures that any effects, particles, sounds, debug draw, whatever presentation logic we want to attach here gets correctly discarded during a correction.